Pa3/Durchführung/Project/app/StateHandler.c

```diff
@@ -31,13 +31,190 @@
+      InitializeMcu_InitializeAll(void); //entry
+      InitializeMcu_DisplayTeamName(void); //exit
+      gCurrentState = STATE_CALIBRATE_LINE_SENSORS;
+      break;
+
+    case STATE_CALIBRATE_LINE_SENSORS:
+      if(gEntryDone == false)
+      {
+        gEntryDone = true;
+        gCurrentEvent = CalibrateLineSensors_Initialize(void); //entry
+        if (NO_EVENT_HAS_HAPPEND != gCurrentEvent)
+        {
+          gCurrentState = STATE_ERROR_HANDLER;
+        }
+      }
+
+      gCurrentEvent = CalibrateLineSensors_CalibrateSensors(void); //do
+      switch (gCurrentEvent)
+      {
+        case CALIBRATION_FAILED:
+          gCurrentState = STATE_ERROR_HANDLER;
+          break;
+
+        case CALIBRATION_DONE:
+          gEntryDone = false;
+          gCurrentState = STATE_READY_TO_DRIVE;
+          break;
+
+        default:
+          break; 
+      }
+      break;
+
+    case STATE_READY_TO_DRIVE:         
+      gCurrentEvent = ReadyToDrive_CheckStateOfButtons(void); //do
+
+      switch(gCurrentEvent)
+      {
+        case CALIBRATION_BUTTON_HAS_BEEN_RELEASED:
+          gCurrentState = STATE_CALIBRATE_LINE_SENSORS;
+          break;
+
+        case START_BUTTON_HAS_BEEN_RELEASED:
+          gCurrentState = STATE_DISPLAY_COUNTDOWN;
+          break;
+
+        case PARAM_BUTTON_HAS_BEEN_RELEASED:
+          gCurrentState = STATE_SET_PARAMETERS;
+          break;
+
+        default:
+          break; 
+      }
+      break;
+
+    case STATE_SET_PARAMETERS
+      SetParameters_SetNextParamSet(void); //entry
+      SetParameters_DisplayParamSet(void); //exit
+      gCurrentState = STATE_READY_TO_DRIVE;
+      break;
+
+    case STATE_DISPLAY_COUNTDOWN:      
+      if(gEntryDone == false)
+      {
+        gEntryDone = true;
+        DisplayCountdown_StartCountdown(void); //entry
+      }
+      gCurrentEvent = DisplayCountdown_DisplayCountdown(void); //do
+
+      if (gCurrentEvent == COUNTDOWN_IS_FINISHED)
+      {
+        gEntryDone = false;
+        DisplayCountdown_StopCountdown(void); //exit
+        gCurrentState = STATE_DRIVE_TO_START;
+      }
+      else
+      {
+        //nothing should happen
+      }
+      break;
+
+    case STATE_DRIVE_TO_START:        
+      if (gEntryDone == false)
+      {
+        gEntryDone = true;
+        DriveToStart_StartMotorsAndTimer(void); //entry
+      }
+      gCurrentEvent = DriveToStart_FollowGuideLine(void); //do
+
+      if (gCurrentEvent == DRIVE_TO_START_IS_ACTIVE_FOR_TOO_LONG)
+      {
+        gCurrentState = STATE_ERROR_HANDLER;
+        gEntryDone = false;
+        DriveToStart_StopTimer(void); //exit
+      }
+      else if (gCurrentEvent == START_FINISH_LINE_WAS_RECOGINZED)
+      {
+        gCurrentState = STATE_DRIVE_TO_FINISH;
+        gEntryDone = false;
+        DriveToStart_StopTimer(void); //exit
+        gLapTimer = DriveToStart_StartTimerAndBeep(void); //State drive to finish is next
+      }
+      else
+      {
+        //nothing should happen
+      }
```

Comment:
this would be a great opportunity for a static function, to make the switch-case more readable
Handle DriveToStartEvent(gCurrentEvent)

Please check your switch case for other parts that could be done in an additinal function. this will make later testing/debugging easier